build.gradle

```diff
@@ -215,6 +215,37 @@ subprojects {
         ruleSetFiles = files(rootProject.file('tools/gradle/pmd/rules.xml'))
     }
 
+    jacoco {
+        toolVersion = "0.8.7"
+    }
+
+    jacocoTestReport {
+        dependsOn test
+        reports {
+            html.required = true
+            xml.required = true
+            csv.required = false
+        }
+    }
+
+    jacocoTestCoverageVerification {
+        violationRules {
+            failOnViolation = false
+            rule {
+                element = 'CLASS'
+                excludes = ['**/*Test*', '**/generated*']
```

Comment:
We may need to revisit this.  In cloud, we had to change the location of the generated sources to consistently use the same location to make this effective.  For now, this is fine as an experiment to see how much of the generated source gets excluded.